### PR TITLE
Report : Batch Item Expiry Status

### PR DESCRIPTION
**Pull request by revant**
- Select From Date and To Date
- Lists Item Batches about to expire or already expired
- Expiry Status column shows number of days in which item/batch will expire

batch_item_expiry_status

### CODE REVIEW

File: erpnext/stock/report/batch_item_expiry_status/batch_item_expiry_status.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+		expiry_date_unicode = frappe.db.get_value('Batch', d.batch_no, 'expiry_date')
+		qty_dict.expires_on = expiry_date_unicode
+
+		exp_date = frappe.utils.data.getdate(expiry_date_unicode)
+		qty_dict.expires_on = exp_date
+
+		expires_in_days = (exp_date - frappe.utils.datetime.date.today()).days
+
+		if expires_in_days > 0:
+			qty_dict.expiry_status = "Expires in {} day(s)".format(expires_in_days)
```
Comment (rmehta) [inline erpnext/stock/report/batch_item_expiry_status/batch_item_expiry_status.py]: translatable ? `_(""Expires in {} day(s)")`

**Review by rmehta — CHANGES_REQUESTED** [review]

File: erpnext/stock/report/batch_item_expiry_status/batch_item_expiry_status.py (hunk 1 above)
Comment (revant) [inline erpnext/stock/report/batch_item_expiry_status/batch_item_expiry_status.py]: columnrename

How about renaming the column to (already translatable) "Expiry (In Days)" and data as plain int?

Comment (rmehta) [inline erpnext/stock/report/batch_item_expiry_status/batch_item_expiry_status.py]: Cool that works better!

Btw there is a better way using `get_columns` using JSON 

https://github.com/frappe/erpnext/blob/develop/erpnext/accounts/report/profitability_analysis/profitability_analysis.py#L121

**Review by nabinhait — APPROVED** [review]

**Comment by mdwala** [comment]
Isn't available quantity per batch is most required field for all?

**Pull request merged**